NetSDK/Services/Client/Classes/SplitClient.cs

```diff
@@ -75,5 +76,48 @@ private void RecordStats(string key, string feature, long start, string treatmen
                 treatmentLog.Log(key, feature, treatment, start);
             }
         }
+
+        private void RecordStats(Key key, string feature, long start, string treatment, string operation, Stopwatch clock)
+        {
+            if (metricsLog != null)
+            {
+                metricsLog.Time(SdkGetTreatment, clock.ElapsedMilliseconds);
+            }
+
+            if (treatmentLog != null)
+            {
+                treatmentLog.Log(key, feature, treatment, start);
+            }
+        }
+
+
+        public string GetTreatment(Domain.Key key, string feature, Dictionary<string, object> attributes = null)
+        {
+            try
+            {
+                var split = splitCache.GetSplit(feature);
+
+                if (split == null)
+                {
+                    Log.Warn(String.Format("Unknown or invalid feature: {0}", feature));
+                    return Control;
+                }
+
+                long start = CurrentTimeHelper.CurrentTimeMillis();
```

Comment:
also, isn't all this code repeated? If I'm not mistaken the only difference should be in the line #110. Correct me if I'm wrong.